src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.Unix.cs

```diff
@@ -48,24 +51,32 @@ private static string GetUtcFullDisplayName(string timeZoneId, string standardDi
         }
 #pragma warning restore IDE0060
 
-        private static void GetStandardDisplayName(string timeZoneId, ref string? displayName)
+        private static CultureInfo UICulture
         {
-            // Determine the culture to use
-            CultureInfo uiCulture = CultureInfo.CurrentUICulture;
-            if (uiCulture.Name.Length == 0)
-                uiCulture = CultureInfo.GetCultureInfo(FallbackCultureName); // ICU doesn't work nicely with InvariantCulture
+            get
+            {
+                if (_uiCulture == null)
+                {
+                    // Determine the culture to use
```

Comment:
I think will be good if we assert here we are not in globalization invariant mode.